File: src/cxx-dev/LiENa/lienaStructure/lienaDatagramStructure/lienaInputQueue.cpp
#include "lienaInputQueue.h"

lienaInputQueue::lienaInputQueue()
{

}

//! ------------------------------------------------------------------------
//!
//! \brief lienaInputQueue::getLength
//! \return
//!
int lienaInputQueue::getLength(){
    int ret = 0;
    mutex.lock();
    ret = inputQueue.size();
    mutex.unlock();
    return ret;
}

//! ---------------------------------------------------------------
//!
//! \brief lienaOutputQueue::clear
//!
void lienaInputQueue::clear(){
    mutex.lock();
    this->inputQueue.clear();
    mutex.unlock();
}

//! ------------------------------------------------------------------------
//!
//! \brief lienaInputQueue::append
//! \param byteArray
//!
void lienaInputQueue::append(lienaDatagram *byteArray){
    mutex.lock();
    inputQueue.append(byteArray);
    mutex.unlock();
}

//! ------------------------------------------------------------------------
//!
//! \brief lienaInputQueue::getFrontDatagram
//! \return
//!
lienaDatagram *lienaInputQueue::getFrontDatagram(){
    lienaDatagram *ret = NULL;
    mutex.lock();
    ret = inputQueue.at(0);
    mutex.unlock();
    return ret;
}

//! ------------------------------------------------------------------------
//!
//! \brief lienaInputQueue::deleteLatestArray
//!
void lienaInputQueue::deleteFrontDatagram(){
    mutex.lock();
    inputQueue.pop_front();
    mutex.unlock();
}

//! ------------------------------------------------------------------------
//!
//! \brief lienaInputQueue::print
//!
void lienaInputQueue::print(){
//    for(int cpt = 0; cpt < inputQueue.size(); cpt++){
//        int array_size = inputQueue.at(cpt).size();

//        qDebug()<<"size:"<<array_size<<", value:"
//                <<(unsigned char)inputQueue.at(cpt).at(0)
//                <<(unsigned char)inputQueue.at(cpt).at(1)
//                <<(unsigned char)inputQueue.at(cpt).at(2)
//                <<(unsigned char)inputQueue.at(cpt).at(3)
//                <<(unsigned char)inputQueue.at(cpt).at(4)
//                <<(unsigned char)inputQueue.at(cpt).at(5)
//                <<(unsigned char)inputQueue.at(cpt).at(6)
//                <<(unsigned char)inputQueue.at(cpt).at(7)
//                <<(unsigned char)inputQueue.at(cpt).at(8)
//                <<(unsigned char)inputQueue.at(cpt).at(9);

//    }
}

File: src/pyDev/LiENa/LiENaStructure/LiENaDatagramStructure/LienaInputQueue.py
import threading


class LienaInputQueue:
    def __init__(self):
        self.inputQueueLock = threading.Lock()
        self.inputQueue = list()

    def append(self, datagram):
        self.inputQueue.append(datagram)
        
    def get_latest_array(self):
        ret = None
        self.inputQueueLock.acquire()
        if len(self.inputQueue) > 0:
            ret = self.inputQueue.pop(0)
        self.inputQueueLock.release()
        return ret
        
    def get_length(self):
        length = 0
        self.inputQueueLock.acquire()
        length = len(self.inputQueue)
        self.inputQueueLock.release()
        return length

    def clear(self):
        self.inputQueueLock.acquire()
        self.inputQueue = []
        self.inputQueueLock.release()
